packages/odyssey-react/src/components/TextInput/TextInput.tsx

```diff
@@ -143,47 +164,54 @@ export const TextInput = withTheme(
           }
         : {};
 
-    const input = (
-      <input
-        {...omitProps}
-        {...ariaProps}
-        className={styles.root}
-        disabled={disabled}
-        id={oid}
-        name={name}
-        onChange={handleChange}
-        onBlur={onBlur}
-        onFocus={onFocus}
-        placeholder={placeholder}
-        readOnly={readonly}
-        ref={ref}
-        required={required}
-        type={type}
-        defaultValue={defaultValue}
-        value={value}
-      />
-    );
-
-    const search = (
-      <span className={styles.outer}>
-        {input}
-        <span className={styles.indicator} role="presentation">
-          <SearchIcon />
-        </span>
-      </span>
-    );
-
     return (
       <Field
         error={error}
         hint={hint}
         inputId={oid}
         label={label}
-        labelHidden={type === "search"}
+        labelHidden={isSearchTextInput}
         optionalLabel={optionalLabel}
         required={required}
       >
-        {type === "search" ? search : input}
+        <div className={styles.root}>
+          {(prefix || isSearchTextInput) && (
+            <span
+              className={styles.prefix}
+              aria-hidden="true"
+              onClick={setFocus}
+            >
+              {isSearchTextInput ? <SearchIcon /> : prefix}
+            </span>
+          )}
+          <input
+            {...omitProps}
+            {...ariaProps}
+            className={styles.input}
+            disabled={disabled}
+            id={oid}
+            name={name}
+            onChange={handleChange}
+            onBlur={onBlur}
+            onFocus={onFocus}
+            placeholder={placeholder}
+            readOnly={readonly}
+            ref={internalRef}
+            required={required}
+            type={type}
+            defaultValue={defaultValue}
+            value={value}
+          />
+          {suffix && (
```

Comment:
I believe in some cases such as `type="search"` we want to prevent adding a suffix too? cc @edburyenegren-okta @josesolano-okta 

Something we can cleanup in a subsequent PR no big deal

Reply:
correct, search will have a trailing button once you type to remove the text but should never have trailing text

Reply:
+1

The trailing button and associated behavior will be added in [OKTA-479985](https://oktainc.atlassian.net/browse/OKTA-479985), and we won't want suffix being used with search in the meantime.

Reply:
Got it, I'll disallow `suffix` it `type="search"`